### PR TITLE
conditional instance restarts on config file changes

### PR DESCRIPTION
**Pull request by aamerik**
Allow conditional restarts of a redis instance on configuration changes.

### CODE REVIEW

File: manifests/instance.pp [hunk 1]
```diff
@@ -356,7 +359,11 @@
       $real_service_ensure = $service_ensure == 'running'
       $real_service_enable = $service_enable
 
-      Exec["cp -p ${redis_file_name_orig} ${redis_file_name}"] ~> Service["${service_name}.service"]
+      if $notify_service {
+        Exec["copy ${redis_file_name_orig} to ${redis_file_name}"] ~> Service["${service_name}.service"]
+      } else {
+        Exec["copy ${redis_file_name_orig} to ${redis_file_name}"] -> Service["${service_name}.service"]
+      }
```
Comment (root-expert) [inline manifests/instance.pp:366]: Is this needed?

File: manifests/instance.pp [hunk 2]
```diff
@@ -89,6 +89,8 @@
 #   If you have latency problems turn this to 'true'. Otherwise leave it as
 # @param notify_keyspace_events
 #   Which events to notify Pub/Sub clients about events happening
+# @param notify_service
+#   You may disable service reloads when config files change if you
```
Comment (root-expert) [inline manifests/instance.pp:93]: Something's missing here :smile:

File: manifests/instance.pp (hunk 1 above)
Comment (aamerik) [reply manifests/instance.pp:366]: I think it still makes sense to set up this dependency chaining for config file to exist prior starting the instance.

**Comment by aamerik** [comment]
Any chance this could get merged? Thanks

**Review by smortex — APPROVED** [review]

**Review by ekohl — APPROVED** [review]

**Pull request merged**